cmds/upspin_sos/service.go

```diff
@@ -78,9 +78,22 @@ func (us UpspinService) setFileData(path string) error {
 }
 
 func (us UpspinService) setKeys(path string) error {
-	// execute as user. This command generates files with elevated permissions otherwise
-	keygen := exec.Command("upspin", "keygen", fmt.Sprintf("-secretseed=%v", us.Seed), path)
-	err := keygen.Run()
+	// check if keys are set already
+	f, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.Readdir(1)
+	// if the directory is populated, rotate the keys instead of generating new ones
+	if err == nil {
```

Comment:
I would do this slightly differently. 
You might do this:
keygen := exec.Command("upspin", "keygen", fmt.Sprintf("-secretseed=%v", us.Seed), path)
if err == nil {
keygen.Args = append(keygen.Args, "-rotate")
}
then very little else needs to change.

Reply:
The command doesn't like putting flags after the path. I could look into making an array of args and stick it at the beginning if it needs to be set?

Reply:
yes, you can do this then.
args := []string{"upspin", "keygen", fmt.Sprintf("-secretseed=%v", us.Seed}
if err == nil {
args = append(args, "-rotate")
}
args = append(args, path)

keygen := exec.Command(args...)
something like that.